### PR TITLE
Append initial worker upstreams if current upstreams are unresponsive

### PR DESCRIPTION
**Pull request by irenarindos**

### CODE REVIEW

File: internal/tests/cluster/multi_controller_worker_test.go [hunk 1]
```diff
@@ -83,3 +85,42 @@ func TestMultiControllerMultiWorkerConnections(t *testing.T) {
...
+
+	time.Sleep(10 * time.Second)
+	expectWorkers(t, c1, w1)
+
+	// Upstreams should be equivalent to the controller cluster addr after status updates
+	assert.Equal(c1.ClusterAddrs(), w1.Worker().LastStatusSuccess().LastCalculatedUpstreams)
+
+	// Bring down the controller
+	c1.Shutdown()
+	time.Sleep(17 * time.Second) // Wait a little longer than the grace period
```
[the remark this answers is not in the document]
Comment (irenarindos) [reply internal/tests/cluster/multi_controller_worker_test.go:122]: Thanks!

File: internal/tests/cluster/multi_controller_worker_test.go [hunk 2]
```diff
@@ -83,3 +85,42 @@ func TestMultiControllerMultiWorkerConnections(t *testing.T) {
...
+
+	initialUpstreams := append(c1.ClusterAddrs(), "127.0.0.9")
+	w1 := worker.NewTestWorker(t, &worker.TestWorkerOpts{
+		WorkerAuthKms:    c1.Config().WorkerAuthKms,
+		InitialUpstreams: initialUpstreams,
+		Logger:           logger.Named("w1"),
+	})
+	defer w1.Shutdown()
+
+	time.Sleep(10 * time.Second)
```
[the remark this answers is not in the document]
Comment (irenarindos) [reply internal/tests/cluster/multi_controller_worker_test.go:114]: ty!

File: internal/daemon/worker/status.go [hunk 3]
```diff
@@ -330,6 +327,42 @@ func (w *Worker) sendWorkerStatus(cancelCtx context.Context, sessionManager sess
 	}
 }
 
+// Update address receivers and dialing listeners with new addrs
+func (w *Worker) updateAddresses(cancelCtx context.Context, addrs []string, addressReceivers *[]addressReceiver) {
+	const op = "worker.(Worker).updateAddrs"
+
+	if len(addrs) > 0 {
```
[the remark this answers is not in the document]
Comment (irenarindos) [reply internal/daemon/worker/status.go:334]: I'll open a Jira ticket for this; looks like we want to group them under a different interface

[the remark this answers is not in the document]
Comment (jefferai) [reply internal/daemon/worker/status.go:334]: Why not just only call it once like the previous code?

[the remark this answers is not in the document]
Comment (jefferai) [reply internal/daemon/worker/status.go:334]: oh nm, the issue isn't update address being called twice, it's setaddress being called twice *in* the function

[the remark this answers is not in the document]
Comment (irenarindos) [reply internal/daemon/worker/status.go:334]: FWIW: https://github.com/hashicorp/boundary-enterprise/pull/189

[the remark this answers is not in the document]
Comment (jefferai) [reply internal/daemon/worker/status.go:334]: Feels like this logic could be combined so SetAddress is called only once.

**Pull request merged**